demux/demux_lavf.c

```diff
@@ -525,6 +525,10 @@ static int lavf_check_file(demuxer_t *demuxer, enum demux_check check)
 
     av_free(avpd.buf);
 
+    // avif always needs to find stream info
+    if (bstrcmp(ext, bstr0("avif")) == 0)
```

Comment:
shouldn't this be an entry in format_hacks array? If not who will restore skipinfo value to previous one? Also format_hack may be null here if there is no coresponding format hack.

Reply:
`use_stream_ids` applies here. `no_pcm_seek` is irrelevant naturally. I kept this out of `format_hacks` because I didn't want to make it order-dependent (i.e. `avif` has to come before `mp4`) and match via the extension instead of the inputformat name.

Reply:
We explicitly skip passing filename https://github.com/mpv-player/mpv/blob/master/demux/demux_lavf.c#L472 to avoid extension dependency of the files that we open. I don't feel it to add a avif exception. All other formats that goes under mov demuxer (`mov,mp4,m4a,3gp,3g2,mj2`) are not affected? Is `skipinfo` needed in general?